api/client/client.go

```diff
@@ -0,0 +1,156 @@
+package client
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	logging "github.com/ipfs/go-log/v2"
+	"google.golang.org/grpc"
+
+	"github.com/celestiaorg/celestia-node/blob"
+	blobapi "github.com/celestiaorg/celestia-node/nodebuilder/blob"
+	blobstreamapi "github.com/celestiaorg/celestia-node/nodebuilder/blobstream"
+	fraudapi "github.com/celestiaorg/celestia-node/nodebuilder/fraud"
+	headerapi "github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/p2p"
+	shareapi "github.com/celestiaorg/celestia-node/nodebuilder/share"
+	stateapi "github.com/celestiaorg/celestia-node/nodebuilder/state"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+var log = logging.Logger("celestia-client")
+
+// Config holds configuration for the Client.
+type Config struct {
+	// TODO: Do we need backwards compatibility tracking? APIVersion will check version of the API and
+	// fail if it doesn't match
+	// APIVersion        string
```

Comment:
i would also not do any API versioning here as the underlying api's themselves not have versioning and just versioning this client does not make sense.